### PR TITLE
fix(closeWindowOsx): recreate window when clicked on dock icon

### PR DESCRIPTION
**Pull request by zckrs**
Sorry if I miss something. 
But actually when we close window the process always running and we can't reopen window.

In first solution we can standardize the behavior across platforms.

In second time we can provide method to restore window on OSX but i didn't find how to do that.

### CODE REVIEW

**Comment by sindresorhus** [comment]
This is probably the event you're looking for: https://github.com/atom/electron/blob/master/docs/api/app.md#event-activate-with-no-open-windows

**Comment by zckrs** [comment]
Thank you. 
I amend commit.

File: app/templates/index.js [hunk 1]
```diff
@@ -29,3 +29,17 @@ app.on('ready', function () {
 		mainWindow = null;
 	});
 });
+
+function createMainWindow () {
+	var browserWin;
+
+	browserWin = new BrowserWindow({
```
Comment (sindresorhus) [inline app/templates/index.js]:
``` js
let browserWin = new BrowserWindow({
```

File: app/templates/index.js [hunk 2]
```diff
@@ -29,3 +29,17 @@ app.on('ready', function () {
 		mainWindow = null;
 	});
 });
+
+function createMainWindow () {
```
Comment (sindresorhus) [inline app/templates/index.js]: put this function near the top of the file

**Comment by sindresorhus** [comment]
Found an issue:

Start the app, close the window, click the dock icon, window opens again, close the window, click the dock icon, no new window shows up this time.

**Comment by zckrs** [comment]
Thank you for reveiw. I will fix.

Do you plan yo use ES6 with Babel ?

**Comment by sindresorhus** [comment]
Yes, just waiting for https://github.com/paulcbetts/electron-compile to be useable so I don't have to manually set up a compile step.

File: app/templates/index.js [hunk 3]
```diff
@@ -5,6 +5,24 @@ const BrowserWindow = require('browser-window');
...
 
+function createMainWindow () {
+	let browserWin = new BrowserWindow({
+		width: 600,
+		height: 400,
+		resizable: false
+	});
+
+	browserWin.loadUrl(`file://${__dirname}/index.html`);
+
```
Comment (sindresorhus) [inline app/templates/index.js:17]: maybe register the `mainWindow.on('closed', onClosed);` here so you don't have to do it twice?

File: app/templates/index.js [hunk 4]
```diff
@@ -14,18 +32,16 @@ app.on('window-all-closed', function () {
...
 });
 
-app.on('ready', function () {
-	mainWindow = new BrowserWindow({
-		width: 600,
-		height: 400,
-		resizable: false
-	});
+app.on('activate-with-no-open-windows', function () {
+	console.log(mainWindow);
```
Comment (sindresorhus) [inline app/templates/index.js]: :zap:

File: app/templates/index.js (hunk 3 above)
Comment (zckrs) [inline app/templates/index.js:17]: Done

File: app/templates/index.js (hunk 4 above)
Comment (zckrs) [inline app/templates/index.js]: :anguished:

**Comment by sindresorhus** [comment]
Thanks :) :beers:

**Comment by zckrs** [comment]
You are welcome.

**Comment by sindresorhus** [comment]
Released as 1.3.0 with some additional improvements ;)

**Pull request closed without merging**